addons/l10n_it/models/account_tax.py

```diff
@@ -45,22 +35,21 @@ class AccountTax(models.Model):
         ],
         string="Exoneration",
         help="Exoneration type",
-        default="N1",
+        default=False,
     )
     l10n_it_law_reference = fields.Char(string="Law Reference", size=100)
 
-    @api.constrains('l10n_it_has_exoneration',
-                    'l10n_it_kind_exoneration',
+    @api.constrains('l10n_it_exempt_reason',
                     'l10n_it_law_reference',
                     'amount',
                     'invoice_repartition_line_ids',
                     'refund_repartition_line_ids')
     def _l10n_it_edi_check_exoneration_with_no_tax(self):
         for tax in self:
-            if tax.l10n_it_has_exoneration:
-                if not tax.l10n_it_kind_exoneration or not tax.l10n_it_law_reference or tax.amount != 0:
-                    raise ValidationError(_("If the tax has exoneration, you must enter a kind of exoneration, a law reference and the amount of the tax must be 0.0."))
-                if tax.l10n_it_kind_exoneration == 'N6' and tax._l10n_it_is_split_payment():
+            if tax.country_id.code == 'IT':
+                if (tax.amount_type == 'percent' and tax.amount == 0) and not (tax.l10n_it_exempt_reason and tax.l10n_it_law_reference):
+                    raise ValidationError(_("If the tax amount is 0%, you must the exoneration code and the related law reference."))
```

Comment:
you must what with the exoneration code? 
```suggestion
                    raise ValidationError(_("If the tax amount is 0%, you must enter the exoneration code and the related law reference."))
```